src/plugins/advanced_settings/public/management_app/advanced_settings.tsx

```diff
@@ -257,10 +262,15 @@ export class AdvancedSettingsComponent extends Component<
 }
 
 export const AdvancedSettings = (props: AdvancedSettingsProps) => {
-  const { query } = useParams<{ query: string }>();
+  const query = useParams<{ query: string }>().query ?? '';
+  const { push } = useHistory();
+  const location = useLocation();
+  const setUrlQuery = (q: string = '') => push({ ...location, pathname: q.toLowerCase() });
```

Comment:
I think we should simplify it and use more official way to do the same
1) pass **history** object from `management_app/mount_management_section.tsx `
2)
```
  const setUrlQuery = useCallback(
    (q: string = '') => {
      props.history.push({
        pathname: q.toLowerCase(),
      });
    },
    [props.history]
  );
```

Reply:
Good point see 336f39e